tmf/org.eclipse.tracecompass.tmf.core/src/org/eclipse/tracecompass/tmf/core/trace/TmfTrace.java

```diff
@@ -304,14 +307,34 @@ public IStatus refreshAnalysisModules() {
 
     @Override
     public IAnalysisModule getAnalysisModule(String analysisId) {
-        return fAnalysisModules.get(analysisId);
+        IAnalysisModule module = fAnalysisModules.get(analysisId);
+        if (module != null) {
+            return module;
+        }
+        return fAddedAnalysisModules.get(analysisId);
     }
 
 
     @Override
     public Iterable<IAnalysisModule> getAnalysisModules() {
         synchronized (fAnalysisModules) {
-            return new HashSet<>(fAnalysisModules.values());
+            HashSet<IAnalysisModule> modules = new HashSet<>(fAnalysisModules.values());
+            modules.addAll(fAddedAnalysisModules.values());
+            return modules;
+        }
+    }
+
+    @Override
+    public IAnalysisModule addAnalysisModule(@NonNull IAnalysisModule module) {
+        synchronized (fAnalysisModules) {
```

Comment:
Do we need a check against fAnalysisModules?

Reply:
Yes, I think it makes sense. I'll throw an exception in case a built-in module exists with the same ID.